components/helm-broker/internal/controller/addons/repository.go

```diff
@@ -0,0 +1,38 @@
+package addons
+
+import (
+	addonsv1alpha1 "github.com/kyma-project/kyma/components/helm-broker/pkg/apis/addons/v1alpha1"
+)
+
+// RepositoryController is a wraper for StatusRepository
+type RepositoryController struct {
+	Repository addonsv1alpha1.StatusRepository
+	Addons     []*AddonController
+}
+
+// NewAddonsRepository returns pointer to new RepositoryController with url and ready status
+func NewAddonsRepository(url string) *RepositoryController {
+	return &RepositoryController{
+		Repository: addonsv1alpha1.StatusRepository{
+			URL:    url,
+			Status: addonsv1alpha1.RepositoryStatusReady,
+		},
+	}
+}
+
+// Failed sets StatusRepository as failed
+func (ar *RepositoryController) Failed() {
+	ar.Repository.Status = addonsv1alpha1.RepositoryStatusFailed
+}
+
+// IsFailed checks is StatusRepository is in failed state
+func (ar *RepositoryController) IsFailed() bool {
+	return ar.Repository.Status == addonsv1alpha1.RepositoryStatusFailed
+}
+
+// FetchingError sets StatusRepository as failed with URLFetchingError as a reason
+func (ar *RepositoryController) FetchingError(err error) {
+	ar.Repository.Status = addonsv1alpha1.RepositoryStatusFailed
+	ar.Repository.Reason = addonsv1alpha1.RepositoryURLFetchingError
+	ar.Repository.Message = err.Error()
```

Comment:
It should be something like this:
```go
reason := addonsv1alpha1.RepositoryURLFetchingError
ar.Repository.Reason = reason
ar.Repository.Message = fmt.Sprintf(reason.String(), err.Error())
```
like in `AddonController` method, thanks that we do not lose [this part of message](https://github.com/kyma-project/kyma/blob/master/components/helm-broker/pkg/apis/addons/v1alpha1/reason.go#L50)